src/FsCodec/Codec.fs

```diff
@@ -24,35 +25,31 @@ type Codec =
 
     /// Generate a <code>IUnionEncoder</code> Codec suitable using the supplied <c>encode</c> and <c>tryDecode</code> functions to map to/from UTF8.
     /// <c>mapCausation</c> provides metadata generation and correlation/causationId mapping based on the Context passed to the encoder
-    static member Create<'Context,'Union>
+    static member Create<'Context,'Union,'Format>
         (   /// Maps from the TypeShape <c>UnionConverter</c> <c>'Contract</c> case the Event has been mapped to (with the raw event data as context)
             /// to the representation (typically a Discriminated Union) that is to be presented to the programming model.
-            tryDecode : FsCodec.ITimelineEvent<byte[]> -> 'Union option,
+            tryDecode : FsCodec.ITimelineEvent<'Format> -> 'Union option,
             /// Maps a fresh Event resulting from a Decision in the Domain representation type down to the TypeShape <c>UnionConverter</c> <c>'Contract</c>
             /// The function is also expected to derive
             ///   a <c>meta</c> object that will be serialized with the same settings (if it's not <c>None</c>)
             ///   and an Event Creation <c>timestamp</c>.
-            encode : 'Union -> string * byte[] * DateTimeOffset option,
+            encode : 'Union -> string * 'Format * DateTimeOffset option,
             /// Uses the 'Context passed to the Encode call and the 'Meta emitted by <c>down</c> to a) the final metadata b) the <c>correlationId</c> and c) the correlationId
-            mapCausation : 'Context option * 'Union -> byte[] * string * string,
-            /// Configuration to be used by the underlying <c>Newtonsoft.Json</c> Serializer when encoding/decoding. Defaults to same as <c>Settings.Create()</c>
-            ?settings,
-            /// Enables one to fail encoder generation if union contains nullary cases. Defaults to <c>false</c>, i.e. permitting them
-            ?rejectNullaryCases)
-        : FsCodec.IUnionEncoder<'Union,byte[],'Context> =
+            mapCausation : 'Context option * 'Union -> 'Format * string * string)
+        : FsCodec.IUnionEncoder<'Union,'Format,'Context> =
         let encode (context,union) =
             let et, d, t = encode union
             let m, correlationId, causationId = mapCausation (context, union)
             et, d, m, correlationId, causationId, t
         Codec.Create(encode,tryDecode)
 
     /// Generate a <code>IUnionEncoder</code> Codec using the supplied pair of <c>encode</c> and <c>tryDecode</code> functions.
-    static member Create<'Union>
+    static member Create<'Union,'Format when 'Format : null>
```

Comment:
Is this function indicating that Format needs to be some nullable type?

Reply:
there's a comparison or defaulting to null somewhere in the chain which means the rendered form must be a reference type